variables.tf

```diff
@@ -0,0 +1,76 @@
+
+
+variable "agent_count" {
+  default = 3
+}
+
+variable "cluster_name" {
+  type        = string
+  default     = null
+  description = "The Domino cluster name for the K8s cluster and resource group"
+}
+
+variable "location" {
+  default = "West US 2"
+}
+
+variable "log_analytics_workspace_name" {
+  default = "testLogAnalyticsWorkspaceName"
+}
+
+# refer https://azure.microsoft.com/global-infrastructure/services/?products=monitor for log analytics available regions
+variable "log_analytics_workspace_location" {
+  default = "eastus"
+}
+
+# refer https://azure.microsoft.com/pricing/details/monitor/ for log analytics pricing 
+variable "log_analytics_workspace_sku" {
+  default = "PerGB2018"
+}
+
+variable "node_pools" {
+  type = map(object({
+    vm_size                        = string
+    zones                          = list(string)
+    node_os                        = string
+    taints                         = list(string)
+    cluster_auto_scaling           = bool
+    cluster_auto_scaling_min_count = number
+    cluster_auto_scaling_max_count = number
+  }))
+  default = {
+    compute = {
+      vm_size                        = "Standard_DS2_v2"
+      zones                          = ["1", "2", "3"]
+      node_os                        = "Linux"
+      taints                         = null
+      cluster_auto_scaling           = true
+      cluster_auto_scaling_min_count = 1
+      cluster_auto_scaling_max_count = 4
+    }
+    gpu = {
+      vm_size                        = "Standard_DS2_v2"
+      zones                          = ["1", "2", "3"]
+      node_os                        = "Linux"
+      taints                         = null
+      cluster_auto_scaling           = true
+      cluster_auto_scaling_min_count = 1
```

Comment:
Should probably setup the gpu taints, and set default gpu node pool size to 0 to keep costs down unless it's actually in use.

Reply:
1. Happy to add the taints - don't believe we do this (yet) for GKE either. 
2. I would love to scale it down to zero - sadly (unlike GKE) the minimum node pool size for AKS is 1 (not zero). That was the first thing I tried :(

Reply:
Here's the taints/labels we use from aws:

```
      taints:
        "nvidia.com/gpu": "true"
      labels:
        "dominodatalab.com/node-pool": "default-gpu"
        "nvidia.com/gpu": "true"
```

About the minimum node pool / can't scale to zero bit, I think we should then remove gpu from the default node pools, and just give an example of how to add them. 99% of the time in testing we're not actually using gpus, yet they're expensive, so we should do what we can to only use them as-needed.